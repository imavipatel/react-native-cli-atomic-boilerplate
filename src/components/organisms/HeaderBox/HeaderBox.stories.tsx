import * as React from 'react';
import { storiesOf } from '@storybook/react-native';
import { HeaderBox } from './HeaderBox';
import CenterView from '../../../../storybook/utils/CenterView/CenterView';

storiesOf('HeaderBox', module)
  .addDecorator(story => <CenterView>{story()}</CenterView>)
  .add('default', () => <HeaderBox>Header Box Title</HeaderBox>);
