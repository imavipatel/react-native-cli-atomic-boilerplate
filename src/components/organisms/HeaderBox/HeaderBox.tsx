import React, { ReactNode } from 'react';
import { View } from 'react-native';

import { Header } from '../../molecules';
import { styles } from './HeaderBox.style';

interface HeaderBoxProps {
  children: ReactNode;
}

export const HeaderBox: React.FC<HeaderBoxProps> = props => {
  return (
    <View style={styles.container}>
      <Header>{props.children}</Header>
    </View>
  );
};
