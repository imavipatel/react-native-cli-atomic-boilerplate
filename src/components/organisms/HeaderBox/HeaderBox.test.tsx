import React from 'react';
import { render } from '@testing-library/react-native';
import { HeaderBox } from './HeaderBox';

describe('HeaderBox', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render correctly', () => {
    const tree = render(<HeaderBox>Header Box title</HeaderBox>);

    expect(tree).toMatchSnapshot();
  });
});
