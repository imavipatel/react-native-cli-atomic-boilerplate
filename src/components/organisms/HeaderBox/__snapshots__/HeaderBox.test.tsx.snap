// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBox should render correctly 1`] = `
<View
  style={
    Object {
      "flex": 2,
      "flexDirection": "column",
      "justifyContent": "flex-end",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "minHeight": 45,
        "minWidth": 200,
      }
    }
    testID="header"
  >
    <Text
      style={
        Object {
          "color": "black",
          "fontSize": 30,
          "fontWeight": "600",
        }
      }
      testID="header-text"
    >
      Header Box title
    </Text>
  </View>
</View>
`;
